### PR TITLE
Add compound keys to dat import

### LINKED ISSUE #372: compound keys
I hear that compound keys are desired (on import)

```
dat import -d dataset /path/to/file.csv --key=city,state,zip
```

Concatenated using a separator, and sorted internally in case they're given in scrambled order

cc @talos

### PR DESCRIPTION
**Pull request by karissa**
@maxogden @mafintosh pls review? :)

fixes #372

### CODE REVIEW

**Comment by mafintosh** [comment]
Instead of adding a new flag we could also implement this by adding multiple `--key` flags

```
dat import -k city -k state -k country
```

Using minimist that gets parsed as `argv.key === ['city', 'state', 'country']`

**Comment by karissa** [comment]
If the field in question has a comma in it, would that interfere?

On Wednesday, August 5, 2015, Mathias Buus <notifications@github.com> wrote:

> Instead of adding a new flag we could also implement this by adding
> multiple --key flags
>
> dat import -k city -k state -k country
>
> Using minimist that gets parsed as argv.key === ['city', 'state',
> 'country']
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maxogden/dat/pull/373#issuecomment-127890596>.
>


-- 
Karissa McKelvey
http://karissa.github.io/ <http://karissamck.com>

Send me an encrypted message:
https://keybase.io/karissa <https://keybase.io/karissa/key.asc>

**Comment by karissa** [comment]
I like your interface

On Wednesday, August 5, 2015, Karissa McKelvey <karissa.mckelvey@gmail.com>
wrote:

> If the field in question has a comma in it, would that interfere?
>
> On Wednesday, August 5, 2015, Mathias Buus <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> Instead of adding a new flag we could also implement this by adding
>> multiple --key flags
>>
>> dat import -k city -k state -k country
>>
>> Using minimist that gets parsed as argv.key === ['city', 'state',
>> 'country']
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/maxogden/dat/pull/373#issuecomment-127890596>.
>>
>
>
> --
> Karissa McKelvey
> http://karissa.github.io/ <http://karissamck.com>
>
> Send me an encrypted message:
> https://keybase.io/karissa <https://keybase.io/karissa/key.asc>
>
>

-- 
Karissa McKelvey
http://karissa.github.io/ <http://karissamck.com>

Send me an encrypted message:
https://keybase.io/karissa <https://keybase.io/karissa/key.asc>

**Comment by karissa** [comment]
@mafintosh okay, updated. thoughts?

File: lib/import.js [hunk 1]
```diff
@@ -11,7 +13,17 @@ module.exports = function (db, opts) {
 
   var transform = through.obj(function (obj, enc, next) {
     debug('heres my obj!', obj)
-    var key = obj[opts.key] || obj.key || uuid()
+
+    var key
+    if (typeof opts.key === 'object' && opts.key.length > 1) {
```
Comment (mafintosh) [inline lib/import.js]: `Array.isArray(opts.key)`

**Pull request merged**